src/widgets/survey/model/services/ConstructCompletionsService.ts

```diff
@@ -117,12 +140,30 @@ export class ConstructCompletionsService {
     }: ConstructForIntermediateInput,
     flowName: string,
     appletName: string,
+    evaluatedEndAt: number,
+    availableTo: number | null,
   ) {
+    const logDates = this.getLogDates(evaluatedEndAt, availableTo);
+
     Logger.log(
-      `[ConstructCompletionsService]: Activity "${activityName}|${activityId}" within flow "${flowName}|${flowId}" completed, applet "${appletName}|${appletId}"`,
+      `[ConstructCompletionsService]: Activity "${activityName}|${activityId}" within flow "${flowName}|${flowId}" completed, applet "${appletName}|${appletId}, ${logDates}"`,
     );
   }
 
+  private evaluateEndAt(
+    completionType: CompletionType,
+    availableTo: number | null,
+    isAutocompletion: boolean,
+  ): number {
+    if (!isAutocompletion || !availableTo) {
+      return getNow().getTime();
+    }
+
+    return completionType === 'intermediate'
+      ? availableTo
+      : addMilliseconds(availableTo, DistinguishInterimAndFinishLag).getTime();
```

Comment:
Please elaborate on why we need to add 1 millisecond for Admind and Backend to be able to sort it the right way. Is there a better alternative for it?

Reply:
Alternative: investigate all related code on BE & Admin. We've already spend some time for investigation and for communication (~2h for every person: me, Ihor, Uladzimir) and current approach looks as optimal. But for ideal approach - investigation and possible changes on BE/Admin sides. And considering all these changes on deployment. I don't think it's worth it.

Reply:
@anq83  thank you for the clarification!
Please add a "TODO" comment then. I saw you already added a comment to a variable declaration, so just add a "TODO" note explicitly.

Reply:
added